src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IUsingStatement.cs

```diff
@@ -0,0 +1,648 @@
+                //         /*<bind>*/using (var c1 = new C())
+                Diagnostic(ErrorCode.ERR_NoConvToIDisp, "var c1 = new C()").WithArguments("C").WithLocation(9, 26)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<UsingStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IUsingStatement_InvalidNonDisposableExistingResource()
+        {
+            string source = @"
+using System;
+
+class C
+{
+
+    public static void M1()
+    {
+        var c1 = new C();
+        /*<bind>*/using (c1)
+        {
+            Console.WriteLine(c1.ToString());
+        }/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+IUsingStatement (OperationKind.UsingStatement, IsInvalid) (Syntax: 'using (c1) ... }')
+  Expression: 
+    ILocalReferenceExpression: c1 (OperationKind.LocalReferenceExpression, Type: C, IsInvalid) (Syntax: 'c1')
+  Body: 
+    IBlockStatement (1 statements) (OperationKind.BlockStatement) (Syntax: '{ ... }')
+      IExpressionStatement (OperationKind.ExpressionStatement) (Syntax: 'Console.Wri ... oString());')
+        Expression: 
+          IInvocationExpression (void System.Console.WriteLine(System.String value)) (OperationKind.InvocationExpression, Type: System.Void) (Syntax: 'Console.Wri ... ToString())')
+            Instance Receiver: 
+              null
+            Arguments(1):
+                IArgument (ArgumentKind.Explicit, Matching Parameter: value) (OperationKind.Argument) (Syntax: 'c1.ToString()')
+                  IInvocationExpression (virtual System.String System.Object.ToString()) (OperationKind.InvocationExpression, Type: System.String) (Syntax: 'c1.ToString()')
+                    Instance Receiver: 
+                      ILocalReferenceExpression: c1 (OperationKind.LocalReferenceExpression, Type: C) (Syntax: 'c1')
+                    Arguments(0)
+                  InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                  OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS1674: 'C': type used in a using statement must be implicitly convertible to 'System.IDisposable'
+                //         /*<bind>*/using (c1)
+                Diagnostic(ErrorCode.ERR_NoConvToIDisp, "c1").WithArguments("C").WithLocation(10, 26)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<UsingStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IUsingStatement_InvalidEmptyUsingBlock()
```

Comment:
Perhaps `EmptyUsingResources`. `Block` sounds more like the body.

Reply:
Done.